numba/cuda/tests/cudapy/test_operator.py

```diff
@@ -36,6 +100,137 @@ def test_truediv(self):
+                got = np.zeros(1, dtype=np.bool8)
+                arg1 = np.random.random(1).astype(np.float16)
+                arg2 = (np.random.random(1) * 100).astype(ty)
+
+                kernel[1, 1](got, arg1[0], arg2[0])
+                expected = op(arg1, arg2)
+                self.assertEqual(got[0], expected)
+
+    @skip_unless_cc_53
+    def test_multiple_float16_comparisons(self):
+        functions = (test_multiple_hcmp_1,
+                     test_multiple_hcmp_2,
+                     test_multiple_hcmp_3,
+                     test_multiple_hcmp_4,
+                     test_multiple_hcmp_5)
+        for fn in functions:
+            with self.subTest(fn=fn):
+                compiled = cuda.jit("void(b1[:], f2, f2, f2)")(fn)
+                ary = np.zeros(1, dtype=np.bool8)
+                arg1 = np.float16(2.)
+                arg2 = np.float16(3.)
+                arg3 = np.float16(4.)
+                compiled[1, 1](ary, arg1, arg2, arg3)
+                self.assertTrue(ary[0])
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_fp16_comparison_ptx(self):
+        functions = (simple_fp16_gt, simple_fp16_ge,
+                     simple_fp16_lt, simple_fp16_le,
+                     simple_fp16_eq, simple_fp16_ne)
+        ops = (operator.gt, operator.ge, operator.lt, operator.le,
+               operator.eq, operator.ne)
+        opstring = ('setp.gt.f16', 'setp.ge.f16',
+                    'setp.lt.f16', 'setp.le.f16',
+                    'setp.eq.f16', 'setp.ne.f16')
+        args = (b1[:], f2, f2)
+
+        for fn, op, s in zip(functions, ops, opstring):
+            with self.subTest(op=op):
+                ptx, _ = compile_ptx(fn, args, cc=(5, 3))
+                self.assertIn(s, ptx)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_mixed_fp16_comparison_ptx(self):
+        functions = (simple_fp16_gt, simple_fp16_ge,
+                     simple_fp16_lt, simple_fp16_le,
+                     simple_fp16_eq, simple_fp16_ne)
+        ops = (operator.gt, operator.ge, operator.lt, operator.le,
+               operator.eq, operator.ne)
+        types_convert = (np.int8,)
+
+        opstring = {operator.gt:'setp.gt.f16',
+                    operator.ge:'setp.ge.f16',
+                    operator.lt:'setp.lt.f16',
+                    operator.le:'setp.le.f16',
+                    operator.eq:'setp.eq.f16',
+                    operator.ne:'setp.ne.f16'}
+        for (fn, op), ty in itertools.product(zip(functions, ops),
+                                              types_convert):
+            with self.subTest(op=op):
+                args = (b1[:], f2, from_dtype(ty))
+                ptx, _ = compile_ptx(fn, args, cc=(5, 3))
+                print(ptx)
+                self.assertIn(opstring[op], ptx)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_mixed_fp16_comparison_ptx2(self):
```

Comment:
This might be a more appropriate name:

```suggestion
    def test_mixed_fp16_comparison_promotion_ptx(self):
```

(seeing that `test_mixed_fp16_comparison_promotion_ptx` as opposed to `test_mixed_fp16_comparison_ptx2` has failed gives more information/context)